gateway/ln-gateway/src/lib.rs

```diff
@@ -1617,3 +1860,57 @@ impl Display for PrettyInterceptHtlcRequest<'_> {
         )
     }
 }
+
+#[derive(Error, Debug, Clone, Eq, PartialEq)]
+pub enum SendPaymentError {
+    #[error("The federation id is unknown")]
+    UnknownFederationId,
```

Comment:
We need to include additional debugging information on all of these errors. i.e federation id, contract, id, invoice hash, timeouts, etc